fetch.bs

```diff
@@ -3756,6 +3804,9 @@ the request.
  <a for="fetch params">cross-origin isolated capability</a> is
  <var>crossOriginIsolatedCapability</var>.
 
+ <li><p>If <var>processFetchStart</var> is not null, invoke <var>processFetchStart</var> with
+ <var>fetchParams</var>'s <a for="fetch params">controller</a>.
```

Comment:
Why the change to this rather than returning the controller?

Reply:
The problem was that HTTP fetch is synchronous and returns a response. So changing SW fetch to be asynchronous would require a big refactor and I try to be consistent. Unless there is some await-style language I can use?

Reply:
HTTP fetch always runs in parallel so in principle we can indeed "wait" for something to happen. Or we could split HTTP fetch and invoke the subsequent part from the callback.

However, I think a simpler version could be that we share the fetch controller with the service worker. That way the service worker can observe the state and act accordingly. That would require less refactoring and less data structures. What do you think?

Reply:
That would make things simpler

But it would mean that the SW terminating its internal fetch would be a "side effect" of termination ('when the controller is terminated'), and I tried to make it an explicit call from a termination algorithm where it's clear what calls what. I can go either way

Reply:
I see. I think if we use https://infra.spec.whatwg.org/#abort-when that should be okay. That would also make it clear what the scope is.